pkg/controller/logstash/driver.go

```diff
@@ -88,7 +88,10 @@ func internalReconcile(params Params) (*reconciler.Results, logstashv1alpha1.Log
 		return results.WithError(err), params.Status
 	}
 
-	if err := reconcilePipeline(params, configHash); err != nil {
+	// We intentionally DO NOT pass the configHash here. We don't want to consider the pipeline definitions in the
+	// hash of the config to ensure that a pipeline change does not automatically trigger a restart
+	// of the pod, but allows Logstash's automatic reload of pipelines to take place
+	if err := reconcilePipeline(params); err != nil {
```

Comment:
Should we pass the `configHash` when `config.reload.automatic`equals false?

Reply:
It's a good question. I'm erring on the side of 'no' at the moment, but I think this is something that could change after the technical preview depending on feedback.

My reasoning on this is that the `false` (default) value of non-k8s logstash doesn't react to pipeline changes at all, and to change this semantic to restart logstash completely on pipeline changes feels like very different behaviour.

Thinking about how we could add flexibility, I wonder if we might want to introduce something for ECK here, along the lines of: 

`config.reload.restart_policy: detected_only|all|none`, which would either set `config.reload.automatic: true` for `detected_only`, and `false` for `all` or `none`, passing the `configHash` if the value is `all`, and not if it is `none`.

cc @flexitrev, @roaksoax, @jsvd